app/amity.py

```diff
@@ -0,0 +1,328 @@
+from database import Office as Ofisi, LivingSpace as Hostel, Person as Mtu
+from person import Fellow, Staff
+from rooms import LivingSpace, Office
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+import random
+from database import Base
+
+
+class Amity(object):
+    '''This class contains all rooms and all people in amity'''
+
+    def __init__(self):
+        self.all_rooms = []
+        self.office_rooms = []
+        self.livingspaces = []
+        self.all_people = []
+        self.fellows = []
+        self.staff = []
+        self.allocated = []
+        self.unallocated = []
+
+    def create_room(self, name, type_room):
+        '''this method create rooms.it can create multiple rooms'''
+        if type_room.lower() == 'livingspace':
+            room = LivingSpace(name)
+            self.livingspaces.append(room)
+            self.all_rooms.append(room)
+
+        elif type_room.lower() == 'office':
+            room = Office(name)
+            self.office_rooms.append(room)
+            self.all_rooms.append(room)
+        else:
+            return 'Invalid Room Type'
+
+    def add_person(self, name, role, wants_accomodation='N'):
+        '''this method creates a person and allocats the person a room'''
+        if role == 'fellow':
+            person = Fellow(name)
+            # pick an office at random from the office_list
+            # import pdb; pdb.set_trace()
+            try:
+                randomized_office = random.choice(self.office_rooms)
+                room_status = self.is_room_full(randomized_office)
```

Comment:
if you select a random room from a list of available offices then this check will not be necessary.. just reduce `self.rooms` to a list of only the rooms that are available